test/test.js

```diff
@@ -88,593 +85,519 @@ var wasInvoked = function(entity, sequence, method) {
+  it('Build the /test/resource2/:resource2Id/nested2 endpoints', () => {
     // Create the schema.
-    var Nested2Schema = new mongoose.Schema({
+    const Nested2Schema = new mongoose.Schema({
       resource2: {
         type: mongoose.Schema.Types.ObjectId,
         ref: 'resource2',
         index: true,
-        required: true
+        required: true,
       },
       title: {
         type: String,
-        required: true
+        required: true,
       },
       age: {
-        type: Number
+        type: Number,
       },
       description: {
-        type: String
-      }
+        type: String,
+      },
     });
 
     // Create the model.
-    var Nested2Model = mongoose.model('nested2', Nested2Schema);
+    const Nested2Model = mongoose.model('nested2', Nested2Schema);
 
     // Create the REST resource and continue.
     Resource(app, '/test/resource2/:resource2Id', 'nested2', Nested2Model).rest({
       // Register before/after global handlers.
-      before: function(req, res, next) {
+      before(req, res, next) {
         req.body.resource2 = req.params.resource2Id;
         req.modelQuery = this.model.where('resource2', req.params.resource2Id);
 
         // Store the invoked handler and continue.
         setInvoked('nested2', 'before', req);
         next();
       },
-      after: function(req, res, next) {
+      after(req, res, next) {
         // Store the invoked handler and continue.
         setInvoked('nested2', 'after', req);
         next();
-      }
+      },
     });
-
-    done();
   });
 });
 
-describe('Test single resource CRUD capabilities', function() {
-  var resource = {};
+describe('Test single resource CRUD capabilities', () => {
+  let resource = {};
 
-  it('/GET empty list', function(done) {
-    request(app)
+  it('/GET empty list', () => {
+    return request(app)
       .get('/test/resource1')
       .expect('Content-Type', /json/)
       .expect('Content-Range', '*/0')
       .expect(200)
-      .end(function(err, res) {
+      .then((res) => {
         assert.equal(res.hasOwnProperty('body'), true);
         assert.deepEqual(res.body, []);
-        done(err);
       });
   });
 
-  it('/POST Create new resource', function(done) {
-    request(app)
+  it('/POST Create new resource', () => {
+    return request(app)
       .post('/test/resource1')
       .send({
         title: 'Test1',
-        description: '12345678'
+        description: '12345678',
       })
       .expect('Content-Type', /json/)
       .expect(201)
-      .end(function(err, res) {
+      .then((res) => {
         resource = res.body;
         assert.equal(resource.title, 'Test1');
         assert.equal(resource.description, '12345678');
         assert(resource.hasOwnProperty('_id'), 'Resource ID not found');
-        done(err);
       });
   });
 
-  it('/GET The new resource', function(done) {
-    request(app)
-      .get('/test/resource1/' + resource._id)
+  it('/GET The new resource', () => {
```

Comment:
Good cleanup!  This could also be further minified by doing the following.

```
it('/GET The new resource', () => request(app)
  .get(`/test/resource1/${resource._id}`)
  .expect('Content-Type', /json/)
  .expect(200)
  .then((res) => {
    assert.equal(res.body.title, resource.title);
    assert.equal(res.body.description, resource.description);
    assert.equal(res.body._id, resource._id);
  }
);
```

Reply:
Sure, will add this.